paasta_tools/cli/cmds/spark_run.py

```diff
@@ -386,7 +398,23 @@ def build_and_push_docker_image(args):
     return docker_url
 
 
+def validate_work_dir(s):
+    dirs = s.split(':')
+    if len(dirs) != 2:
+        msg = "--work-dir requires format local_abs_dir:container_abs_dir"
```

Comment:
This can be better. Please add the actual (s) that was provided, to hint at the user "what" needs to change exactly.

The raising, argparse.ArgumentTypeError is a very pythony thing to do, but isn't very user friendly. Does it show a stacktrace to the user?

If so, then please do something else and just exit. We don't need to show a whole scary stacktrace on a "normal" error.